src/FA2.hs

```diff
@@ -0,0 +1,85 @@
+module FA2 
+  ( tokenMetadataRegistry
+  , addOperator
+  ) where
+
+import Text.Show (Show (..))
+
+import Lorentz
+import Lorentz.Contracts.Spec.FA2Interface hiding (name, sender) 
+
+import Util.Markdown
+import Util.Named
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+
+type Operators = BigMap (Address, Address) ()
+
+type Ledger = BigMap (Address, TokenId) Natural
+
+type StorageTokenMetadata = ("token_metadata" :! BigMap TokenId TokenMetadata)
+
+-- | Storage of the FA2 contract
+data Storage = Storage
+  { sLedger      :: Ledger
+  , sOperators   :: Operators
+  , sTokenMetadata :: StorageTokenMetadata
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass (IsoValue, HasAnnotation)
+
+instance TypeHasDoc Storage where
+  typeDocMdDescription =
+    "Contract's storage holding a big_map with all balances and the operators."
+
+tokenMetadataRegistry :: forall store. Entrypoint TokenMetadataRegistryParam store
+tokenMetadataRegistry = do
+  sender
```

Comment:
Why does it return `sender`? Let's store metadata registry contract in the storage like we are doing in `stablecoin`.

Reply:
Can you explain further please

Reply:
This entrypoint returns address of the contract that holds token metadata. It's NOT the sender. We should store this address in the storage and return it here (by passing it to the callback contract).